lib/Test/Nginx/Socket.pm

```diff
@@ -2193,6 +2196,10 @@ sub gen_curl_cmd_from_req ($$) {
         $link = "http://$server:$port$uri";
     }
 
+    if (use_http2($block) == 2) {
+        $link = "https://$ServerName:$ServerPortForClient$uri";
```

Comment:
SSL support need proper server certificate and private keys. And it's a burden for the test writer to prepare such things themselves. We should make HTTP/2 over TLS test mode work on most of the existing test suite written for HTTP/1. So we should automatically generate such self-signed server certificate/key pair ourselves in the test scaffold.